src/ui/Facet/ValueElementRenderer.ts

```diff
@@ -176,32 +138,129 @@ export class ValueElementRenderer {
     $$(elem).on('blur', () => $$(this.listItem).removeClass('coveo-focused'));
   }
+
+  private buildListItem() {
+    this.listItem = $$('li', { className: 'coveo-facet-value coveo-facet-selectable' }).el;
+    this.listItem.setAttribute('data-value', this.facetValue.value);
+  }
+
+  private initAndAppendLabel() {
+    this.label = $$('label', { className: 'coveo-facet-value-label' }).el;
+    this.tryToInitAndAppendComputedField();
+    this.initAndAppendFacetValueLabelWrapper();
+    this.listItem.appendChild(this.label);
+  }
+
+  private initAndAppendExcludeIcon() {
+    this.excludeIcon = this.buildExcludeIcon();
+    this.attachExcludeIconEventHandlers();
+    this.listItem.appendChild(this.excludeIcon);
+  }
+
+  private attachExcludeIconEventHandlers() {
+    $$(this.excludeIcon).on('mouseover', () => {
+      $$(this.listItem).addClass('coveo-facet-value-will-exclude');
+    });
+
+    $$(this.excludeIcon).on('mouseout', () => {
+      $$(this.listItem).removeClass('coveo-facet-value-will-exclude');
+    });
+  }
+
+  private tryToInitAndAppendComputedField() {
+    if (!Utils.exists(this.facetValue.computedField)) {
+      return;
+    }
+
+    this.computedField = this.buildValueComputedField();
+
+    if (!this.computedField) {
+      return;
+    }
+
+    this.label.appendChild(this.computedField);
+    $$(this.label).addClass('coveo-with-computed-field');
+  }
+
+  private initAndAppendFacetValueLabelWrapper() {
+    this.facetValueLabelWrapper = $$('div', { className: 'coveo-facet-value-label-wrapper' }).el;
+
+    this.initAndAppendCheckbox();
+    this.initAndAppendStylishCheckbox();
+    this.initAndAppendValueCount();
+    this.initAndAppendValueCaption();
+
+    this.label.appendChild(this.facetValueLabelWrapper);
+  }
+
+  private initAndAppendCheckbox() {
+    this.checkbox = this.buildValueCheckbox();
+    this.facetValueLabelWrapper.appendChild(this.checkbox);
+  }
+
+  private initAndAppendStylishCheckbox() {
+    this.stylishCheckbox = this.buildValueStylishCheckbox();
+    this.facetValueLabelWrapper.appendChild(this.stylishCheckbox);
+  }
+
+  private initAndAppendValueCount() {
+    this.valueCount = this.buildValueCount();
+
+    if (!this.valueCount) {
+      return;
+    }
+
+    this.facetValueLabelWrapper.appendChild(this.valueCount);
+  }
+
+  private initAndAppendValueCaption() {
+    this.valueCaption = this.buildValueCaption();
+    this.facetValueLabelWrapper.appendChild(this.valueCaption);
+  }
+
+  private addAccessibilityAttributesToTargetElement() {
+    const el = this.accessibleElement;
+    el.setAttribute('aria-label', this.ariaLabel);
+    el.setAttribute('role', 'heading');
+    el.setAttribute('aria-level', '3');
+  }
+
+  private get ariaLabel() {
+    const selectOrUnselect = !this.facetValue.selected ? 'Select' : 'Unselect';
+    return `${l(selectOrUnselect)} ${this.caption} ${l('With')} ${l('ResultCount', this.count)}`;
```

Comment:
This will cause problem when translating to different language : It works in french or english because the phrase structure is similar, but will not work for all languages.

If we need to have this, then it needs to be a "full phrase" that we translate, using parameters in the string.

You can see some examples of this in the strings.json file, where there are strings with `{0}` and `{1}` (parameters that you can pass to the translation).